apps/web/src/app/(profile)/[username]/_components/activity-chart.tsx

```diff
@@ -0,0 +1,167 @@
+'use client';
+import {
+  ChartContainer,
+  type ChartConfig,
+  ChartTooltip,
+  ChartTooltipContent,
+} from '@repo/ui/components/chart';
+import { Scatter, ScatterChart, XAxis, YAxis, ZAxis } from '@repo/ui/recharts';
+import { format, setWeek, startOfYear } from 'date-fns';
+import { MessageCircle, FileCode, Award } from '@repo/ui/icons';
+import { cn } from '@repo/ui/cn';
+import { atom, useAtom, type PrimitiveAtom } from 'jotai';
+import { useMemo } from 'react';
+import { useIsMobile } from '~/utils/useIsMobile';
+
+const chartConfig = {
+  submissions: {
+    label: 'Submissions',
+    icon: FileCode,
+  },
+  comments: {
+    label: 'Comments',
+    icon: MessageCircle,
+  },
+  badges: {
+    label: 'Badges',
+    icon: Award,
+  },
+} satisfies ChartConfig;
+
+export function ActivityChart(props: {
+  data: {
+    date: Date;
+    day: 0 | 1 | 2 | 3 | 4 | 5 | 6;
+    week: number;
+    month: number;
+    comments: number;
+    badges: number;
+    submissions: number;
+    activity: number;
+  }[];
+}) {
+  return (
+    <ChartContainer config={chartConfig} className="pointer-events-auto aspect-[8/7] h-[270px]">
+      <ScatterChart data={props.data} accessibilityLayer>
+        <ChartTooltip
+          cursor={false}
+          content={({ payload, content, ...props }) => {
+            const innerPayload = (payload?.[0]?.payload as Record<string, number>) ?? {};
```

Comment:
do we want to memo this?

Reply:
good callout, lemme try and see if I can prevent unnecessary rerenders here